### PR TITLE
Assert that inference model has been persisted

### PR DESCRIPTION
**Pull request by przemekwitek**
- Assert that inference model has been persisted.
- Un-mute ClassificationIT tests.

If there are problems with persisting the inference model, we will learn that as the new assertion will fail.

Relates #48337

### CODE REVIEW

File: x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/MlNativeDataFrameAnalyticsIntegTestCase.java [hunk 1]
```diff
@@ -173,12 +176,20 @@ protected void assertProgress(String id, int reindexing, int loadingData, int an
...
+        String docId = DataFrameAnalyticsTask.progressDocId(jobId);
         return client().prepareSearch(AnomalyDetectorsIndex.jobStateIndexPattern())
-            .setQuery(QueryBuilders.idsQuery().addIds(DataFrameAnalyticsTask.progressDocId(id)))
+            .setQuery(QueryBuilders.idsQuery().addIds(docId))
             .get();
     }
 
+    protected void assertInferenceModelPersisted(String jobId) {
+        SearchResponse searchResponse = client().prepareSearch(InferenceIndexConstants.LATEST_INDEX_NAME)
+            .setQuery(QueryBuilders.prefixQuery(TrainedModelConfig.MODEL_ID.getPreferredName(), jobId + "-"))
```
Comment (benwtrent) [inline x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/MlNativeDataFrameAnalyticsIntegTestCase.java]: You can use a `term` query against the `tags` field.
Something like the following.
```
QueryBuilders.boolQuery().filter(QueryBuilders.termQuery(TrainedModelConfig.TAGS.getPreferredName(), jobId))
```

**Review by benwtrent — COMMENTED** [review]
Won't there still be issues with the audit messages?

**Review by przemekwitek — COMMENTED** [review]
> Won't there still be issues with the audit messages?

To increase chances there won't be any issues, I've relaxed the assertion (now it only requires the specified audit messages but excessive messages won't fail the assertion).
Let's see how it behaves on CI. I think the most important thing now is to enable ClassificationIT and make it work on CI. We can play with the audit message assertion later on.

**Comment by przemekwitek** [comment]
run elasticsearch-ci/2

**Comment by przemekwitek** [comment]
run elasticsearch-ci/bwc

**Review by benwtrent — APPROVED** [review]

**Pull request merged**